tensorboard/plugins/text_v2/text_v2_plugin_test.py

```diff
@@ -0,0 +1,230 @@
+                    )
+                    ctx = base_plugin.TBContext(
+                        logdir=self.logdir,
+                        multiplexer=multiplexer,
+                        data_provider=provider,
+                    )
+                    fn(self, text_v2_plugin.TextV2Plugin(ctx), *args, **kwargs)
+
+            return wrapper
+
+        return decorator
+
+    def generate_testdata(self, include_text=True, logdir=None):
+        tf.compat.v1.reset_default_graph()
+        sess = tf.compat.v1.Session()
+        placeholder = tf.compat.v1.placeholder(tf.string)
+        summary_tensor = tf.compat.v1.summary.text("message", placeholder)
+        vector_summary = tf.compat.v1.summary.text("vector", placeholder)
+        scalar_summary = tf.compat.v1.summary.scalar("twelve", tf.constant(12))
+
+        run_names = ["fry", "leela"]
+        for run_name in run_names:
+            subdir = os.path.join(self.logdir, run_name)
+            with test_util.FileWriterCache.get(subdir) as writer:
+                writer.add_graph(sess.graph)
+
+                step = 0
+                for gem in GEMS:
+                    message = run_name + " *loves* " + gem
+                    feed_dict = {
+                        placeholder: message,
+                    }
+                    if include_text:
+                        summ = sess.run(summary_tensor, feed_dict=feed_dict)
+                        writer.add_summary(summ, global_step=step)
+                    step += 1
+
+                vector_message = ["one", "two", "three", "four"]
+                if include_text:
+                    summ = sess.run(
+                        vector_summary, feed_dict={placeholder: vector_message}
+                    )
+                    writer.add_summary(summ)
+
+                summ = sess.run(scalar_summary, feed_dict={placeholder: []})
+                writer.add_summary(summ)
+
+    @with_plugin()
+    def testRoutesProvided(self, plugin):
+        routes = plugin.get_plugin_apps()
+        self.assertIsInstance(routes["/tags"], collections.abc.Callable)
+        self.assertIsInstance(routes["/text"], collections.abc.Callable)
+
+    @with_plugin()
+    def testIndex(self, plugin):
+        index = plugin.index_impl(experiment="123")
+        self.assertItemsEqual(["fry", "leela"], index.keys())
+        self.assertItemsEqual(["message", "vector"], index["fry"])
+        self.assertItemsEqual(["message", "vector"], index["leela"])
+
+    @with_plugin()
+    def testText(self, plugin):
+        fry = plugin.text_impl("fry", "message", experiment="123")
+        leela = plugin.text_impl("leela", "message", experiment="123")
+        self.assertEqual(len(fry), 4)
+        self.assertEqual(len(leela), 4)
+        for i in range(4):
+            self.assertEqual(fry[i]["step"], i)
+            self.assertEqual(leela[i]["step"], i)
+
+        table = plugin.text_impl("fry", "vector", experiment="123")[0][
```

Comment:
Rather than calling `text_impl()` several times, a bit more compact to just do it like this:

```python
result = plugin.text_impl("fry", "vector", experiment="123")[0]
self.assertEqual(result["string_array"], [b"one", b"two", b"three", b"four"])
self.assertEqual(result["original_shape"], (4,))
...
```

Also I probably wouldn't name this variable `table` since while that made sense for the old code when it had been converted into a `<table>` already, in this case it's less clear.

Reply:
done